src/NServiceBus.Extensions.Hosting/NServiceBusHostedService.cs

```diff
@@ -4,21 +4,22 @@
     using System.Threading;
     using System.Threading.Tasks;
     using Logging;
+    using Microsoft.Extensions.DependencyInjection;
     using Microsoft.Extensions.Hosting;
     using ILoggerFactory = Microsoft.Extensions.Logging.ILoggerFactory;
 
-    class NServiceBusHostedService : IHostedService
+    class NServiceBusHostedService : IHostedService, IMessageSession
```

Comment:
This becomes a bit of a god class but decoupling those things makes everything so much more complicated from a DI and resolve perspective